python/pyarrow/tests/test_array.py

```diff
@@ -688,6 +688,69 @@ def test_dictionary_from_numpy():
             assert d2[i].as_py() == dictionary[indices[i]]
 
 
+def test_dictionary_to_numpy():
+    expected = pa.array(
+        ["foo", "bar", None, "foo"]
+    ).to_numpy(zero_copy_only=False)
+    a = pa.DictionaryArray.from_arrays(
+        pa.array([0, 1, None, 0]),
+        pa.array(['foo', 'bar'])
+    )
+    assert (a.to_numpy(zero_copy_only=False) == expected).all()
```

Comment:
I am not sure if we are very consistent about it, but we also use `np.testing.assert_array_equal` for asserting numpy arrays

Reply:
I don't have a strong opinion about it, if you want me to change those to `assert_array_equal` I'll gladly do. I'm just used to go for `assert` when using `pytest` but in the end `np.testing` provides a good instrumentation too.

Reply:
`np.testing` will provide more useful error messages.

Reply:
replaced the occurrences of `assert (X == Y).all()` with `np.testing.assert_array_equal`